### PR TITLE
Intro to MVC on Mac

### LINKED ISSUE #3500: Create Mac version of Intro to MVC tutorials
Delete Retire https://docs.microsoft.com/en-us/aspnet/core/tutorials/your-first-mac-aspnet when this is complete. when it's complete and redirect to the the Mac version.

### PR DESCRIPTION
**Pull request by Rick-Anderson**
[Review URL](https://review.docs.microsoft.com/en-us/aspnet/core/tutorials/first-mvc-app-mac/index?branch=pr-en-us-3414) 
Fixes #3500

### CODE REVIEW

File: aspnetcore/includes/mvc-intro/adding-controller2.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+* Uses`HtmlEncoder.Default.Encode` to protect the app from malicious input (namely JavaScript). 
+* Uses [Interpolated Strings](https://docs.microsoft.com/dotnet/articles/csharp/language-reference/keywords/interpolated-strings).
+
+Run your app and browse to:
+
+   `http://localhost:xxxx/HelloWorld/Welcome?name=Rick&numtimes=4`
+
+(Replace xxxx with your port number.) You can try different values for `name` and `numtimes` in  the URL. The MVC [model binding](../../mvc/models/model-binding.md) system automatically maps the named parameters from  the query string in the address bar to parameters in your method. See [Model Binding](../../mvc/models/model-binding.md) for more information.
+
+![Browser window showing an application response of Hello Rick, NumTimes is: 4](../../tutorials/first-mvc-app/adding-controller/_static/rick4.png)
```
Comment (tdykstra) [inline aspnetcore/includes/mvc-intro/adding-controller2.md]: This one is an IE screenshot.

File: aspnetcore/includes/mvc-intro/adding-controller2.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+In the sample above, the URL segment (`Parameters`) is not used, the `name` and `numTimes` parameters are passed as [query strings](http://en.wikipedia.org/wiki/Query_string). The `?` (question mark) in the above URL is a separator, and the query strings follow. The `&` character separates query strings.
```
Comment (tdykstra) [inline aspnetcore/includes/mvc-intro/adding-controller2.md]: above is a screenshot, not a sample

File: aspnetcore/includes/mvc-intro/adding-controller2.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+Replace the `Welcome` method with the following code:
+
+[!code-csharp[Main](../../tutorials/first-mvc-app/start-mvc/sample/MvcMovie/Controllers/HelloWorldController.cs?name=snippet_3)]
+
+Run the app and enter the following URL:  `http://localhost:xxx/HelloWorld/Welcome/3?name=Rick`
+
+![Browser window showing an application response of Hello Rick, ID: 3](../../tutorials/first-mvc-app/adding-controller/_static/rick_routedata.png)
```
Comment (tdykstra) [inline aspnetcore/includes/mvc-intro/adding-controller2.md]: IE screenshot

File: aspnetcore/tutorials/first-mvc-app-mac/start-mvc.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+In Visual Studio, select **Run > Start With Debugging** to launch the app. Visual Studio launches a browser and navigates to `http://localhost:port`, where *port* is a randomly chosen port number.
+
+![Browser with New project](start-mvc/b1.png)
+
+* Visual Studio starts [IIS Express](http://www.iis.net/learn/extensions/introduction-to-iis-express/iis-express-overview) and runs your app. Notice that the address bar shows `localhost:port#` and not something like `example.com`. That's because `localhost` is the standard hostname for your local computer. When Visual Studio creates a web project, a random port is used for the web server. In the image above, the port number is 5000. When you run the app, you'll see a different port number.
+* You can launch the app in debug or non-debug mode from the **Run** menu.
+
+The default template gives you **Home, About** and **Contact** links. The browser image above doesn't show these links. Depending on the size of your browser, you might need to click the navigation icon to show them.
+
+![Browser with New project](start-mvc/b2.png)
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/first-mvc-app-mac/start-mvc.md]: I'll update that next version. Yellow does give better contrast than red.

File: aspnetcore/includes/mvc-intro/adding-controller2.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+An HTTP endpoint is a targetable URL in the web application, such as `http://localhost:1234/HelloWorld`, and combines the protocol used: `HTTP`, the network location of the web server (including the TCP port): `localhost:1234` and the target URI `HelloWorld`.
+
+The first comment states this is an [HTTP GET](http://www.w3schools.com/tags/ref_httpmethods.asp) method that is invoked by appending "/HelloWorld/" to the base URL. The second comment specifies an [HTTP GET](http://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html) method that is invoked by appending "/HelloWorld/Welcome/" to the URL. Later on in the tutorial we'll use the scaffolding engine to generate `HTTP POST` methods.
+
+Run the app in non-debug mode (press Ctrl+F5) and append "HelloWorld" to the path in the address bar. (In the image below, `http://localhost:5000/HelloWorld` is used, but you'll have to replace *5000* with the port number of your app.) The `Index` method returns a string. You told the system to return some HTML, and it did!
+
+![Browser window showing an application response of This is my default action](../../tutorials/first-mvc-app/adding-controller/_static/hell1.png)
```
Comment (Rick-Anderson) [inline aspnetcore/includes/mvc-intro/adding-controller2.md]: #3505 will fix that

File: aspnetcore/includes/mvc-intro/adding-controller2.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+`/[Controller]/[ActionName]/[Parameters]`
+
+You set the format for routing in the *Startup.cs* file.
+
+[!code-csharp[Main](../../tutorials/first-mvc-app/start-mvc/sample/MvcMovie/Startup.cs?name=snippet_1&highlight=5)]
+
+When you run the app and don't supply any URL segments, it defaults to the "Home" controller and the "Index" method specified in the template line highlighted above.
+
+The first URL segment determines the controller class to run. So `localhost:xxxx/HelloWorld` maps to the `HelloWorldController` class. The second part of the URL segment determines the action method on the class. So `localhost:xxxx/HelloWorld/Index` would cause the `Index` method of the `HelloWorldController` class to run. Notice that we only had to browse to `localhost:xxxx/HelloWorld` and the `Index` method was called by default. This is because `Index` is the default method that will be called on a controller if a method name is not explicitly specified. The third part of the URL segment ( `id`) is for route data. We'll see route data later on in this tutorial.
```
Comment (Rick-Anderson) [inline aspnetcore/includes/mvc-intro/adding-controller2.md]: #3506 but I'm fixing them all now and I'll fix them as I got forward.

File: aspnetcore/includes/mvc-intro/adding-controller2.md [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+
+Browse to `http://localhost:xxxx/HelloWorld/Welcome`. The `Welcome` method runs and returns the string "This is the Welcome action method...". For this URL, the controller is `HelloWorld` and `Welcome` is the action method. We haven't used the `[Parameters]` part of the URL yet.
+
+![Browser window showing an application response of This is the Welcome action method](../../tutorials/first-mvc-app/adding-controller/_static/welcome.png)
```
Comment (Rick-Anderson) [inline aspnetcore/includes/mvc-intro/adding-controller2.md]: #3505

File: aspnetcore/includes/mvc-intro/adding-controller2.md (hunk 1 above)
Comment (Rick-Anderson) [inline aspnetcore/includes/mvc-intro/adding-controller2.md]: #3505

**Comment by Rick-Anderson** [comment]
@tdykstra  everything fixed except screenshots and yellow callout. I'll do them in #3505 if that's OK. Great feedback.

**Pull request merged**